### PR TITLE
[Button] Fix width so that it stays the same when loading

### PR DESCRIPTION
**Pull request by LauraAubin**
### WHY are these changes introduced?

Follow up from https://github.com/Shopify/polaris-react/pull/3370 which caused buttons with only icons (like pagination) to have extra spacing to the left of the icon.

### WHAT is this pull request doing?

This PR ensures that all buttons keep the same width when they are loading or not. The loading state renders a spinner which is an icon, this was removing the `margin-left` property from the initial icon since it is longer the first child (spinner becomes the first child instead).

The only button variation I found that still resizes its width is a button that uses the `iconOnly` style (since this sets the margin property to 0) and still has another element (see the 3rd example in the below gif of a button with only an icon and a connected disclosure). Though I believe this variation would never exist in the wild.

Buttons

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>The following snippet was pasted into README.md for the Button:</summary>

```jsx
function button() {
  const [loading, isLoading] = useState(false);

  return (
    <div>
      <Button loading={loading}>Add product</Button>
      <br />
      <br />
      <Button loading={loading} icon={ChevronLeftMinor} />
      <br />
      <br />
      <Button loading={loading} icon={ChevronLeftMinor} disclosure={'up'} />
      <br />
      <br />
      <Button
        loading={loading}
        icon={ChevronLeftMinor}
        connectedDisclosure={{
          accessibilityLabel: 'Other save actions',
          actions: [{content: 'Save as draft'}],
        }}
      />
      <br />
      <Button outline icon={ChevronLeftMinor} loading={loading}>
        Add product
      </Button>
      <br />
      <br />
      <Button loading={loading} disclosure={'up'}>
        Add product
      </Button>
      <br />
      <br />
      <Button icon={ChevronLeftMinor} loading={loading} disclosure={'down'}>
        Add product
      </Button>
      <br />
      <br />
      <Button onClick={() => isLoading(!loading)}>Toggle</Button>
    </div>
  );
}
```

</details>

### 🎩 checklist

* [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
* [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
* [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
* [ ] Updated the component's `README.md` with documentation changes
* [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
* [ ] For visual design changes, pinged one of @ HYPD, @ mirualves, @ sarahill, or @ ry5n to update the Polaris UI kit

### CODE REVIEW

File: src/components/Button/Button.tsx [hunk 1]
```diff
@@ -170,6 +170,7 @@ export function Button({
         className={classNames(
           styles.DisclosureIcon,
           disclosure === 'up' && styles.DisclosureIconFacingUp,
+          loading && styles.Hidden
```
Comment (LauraAubin) [inline src/components/Button/Button.tsx:173]: Missed this one in the previous PR, the disclosure icon was also flashing a black square.

File: src/components/Button/Button.scss [hunk 2]
```diff
@@ -119,11 +119,9 @@ $stacking-order: (
...
   transition: color duration() easing();
 
-  &:first-child {
-    // This compensates for the typical icon used in buttons being
-    // inset within its bounding box.
-    margin-left: -(spacing(extra-tight));
-  }
+  // This compensates for the typical icon used in buttons being
+  // inset within its bounding box.
+  margin-left: -(spacing(extra-tight));
```
Comment (LauraAubin) [inline src/components/Button/Button.scss:124]: I couldn't find a good reason to limit this to the first child. When there are multiple icons in a button it is either a generic icon, the loading spinner, or the disclosure. As shown in my tophatting gif, these 3 cases are accounted for (the last button in the gif is a good example of a button using all 3).

**Comment by alex-page** [comment]
@LauraAubin removed changelog and added skip changelog label as your previous PR hasn't shipped.

**Review by alex-page — APPROVED** [review]
Code looks good, I haven't tophatted yet.

**Comment by alex-page** [comment]
Thank you @LauraAubin I am going to merge this as it's an awesome fix that people want ❤️

**Comment by kyledurand** [comment]
Amazing! Thanks for the quick fix @LauraAubin 🥇

**Pull request merged**